common/unsafe/src/main/java/org/apache/spark/unsafe/types/UTF8String.java

```diff
@@ -1484,17 +1485,17 @@ public static UTF8String concatWs(UTF8String separator, UTF8String... inputs) {
   public UTF8String[] split(UTF8String pattern, int limit) {
     // For the empty `pattern` a `split` function ignores trailing empty strings unless original
     // string is empty.
-    if (numBytes() != 0 && pattern.numBytes() == 0) {
+    if (numBytes() != 0 && (pattern.numBytes() == 0 || lowercaseRegexPrefix.equals(pattern))) {
```

Comment:
how about: instead of checking whether the pattern equals to the (?ui) prefix, we modify the collation implementation (prefixing logic) to avoid appending the prefix at all in the case that pattern is an empty string

Reply:
I agree with what you're saying, but should we consider that the user's pattern itself might be `(?ui)` and is unrelated to prefixing logic?

Reply:
that is an interesting observation, although in that case I don't see why the user's pattern can't be any other flag modifier combination, such as: (?m), (?s), (?x), (?a)

taking this into consideration, there is really nothing special about `lowercaseRegexPrefix`. instead, you should look for a library method that can discern whether a pattern is "functionally" empty, instead of doing a manual check against `lowercaseRegexPrefix`

Reply:
Thank you for your explanation. It looks like there’s no way to validate this "weird" situation without losing performance. I made changes according to your advice. Thanks again.